java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/model/AnalysisResult.java

```diff
@@ -92,11 +98,49 @@ private <T> void formatLog(VersionKey root, Map<VersionKey, List<T>> map, String
   }
 
   private String beginSeparator(VersionKey versionKey, VersionKey root) {
-    return String.format("====================== %s, dependency of %s ======================",
-        versionKey.toString(), root.toString());
+    String relation = versionKey.equals(root) ? "self" : "dependency";
+    return String.format("====================== %s, %s of %s ======================",
+        versionKey, relation, root);
   }
 
   private String endSeparator() {
     return "===========================================================";
   }
+
+  private String packageInfoReport() {
+    StringBuilder builder = new StringBuilder();
+    builder.append("Please copy and paste the package information below to your ticket.\n");
+    builder.append("\n\n");
+    appendToReport(builder, packageInfos.get(0));
+
+    builder.append("## Dependencies:\n");
+    if (packageInfos.size() <= 1) {
+      builder.append("None");
+    } else {
+      for (int i = 1; i < packageInfos.size(); i++) {
+        appendToReport(builder, packageInfos.get(i));
+      }
+    }
+    builder.append("\n\n");
+
+    return builder.toString();
+  }
+
+  private void appendToReport(StringBuilder builder, PackageInfo packageInfo) {
+    VersionKey versionKey = packageInfo.versionKey();
+    // generate the report using Markdown format.
+    builder.append(String.format("## Package information of %s\n", versionKey));
+    builder.append(String.format("Licenses: %s\n", packageInfo.licenses()));
+    builder.append(
+        String.format("Vulnerabilities: None.\nChecked in [deps.dev query](%s)\n",
```

Comment:
The visible part of the link should have the artifact coordinates.

Reply:
Done.